File: Práticas/C++/Folha 2/daa005.cpp
#include <iostream>
#include <cstdio>
using namespace std;

int main(){
  int N, soma = 0, F;
  cin >> N;
  
  int energia[N+1];
  int ac[N+1];

  for(int i=1; i<N+1; i++){
    cin >> energia[i];
    ac[i] = ac[i-1] + energia[i];
  }

  cin >> F;

  for(int j=0; j<F; j++){
    int A, B;

    cin >> A;
    cin >> B;

    for(int k=A; k<=B; k+=soma){
      soma = ac[B] - ac[A-1];
    }

    if(A == B)
      soma = energia[A];

    cout << soma << '\n';
    soma = 0;
  }
  
  return 0;
}

File: Práticas/Java/#02/daa005.java
import java.util.Scanner;

public class daa005{
    public static void main(String[] args){
	FastScanner in = new FastScanner(System.in);
	int N = in.nextInt();
	int soma = 0;
	int energia[] = new int[N+1];
	int ac[] = new int[N+1];
	
	for(int i=1; i<N+1; i++){
	    energia[i] = in.nextInt();
	    ac[i] = ac[i-1] + energia[i];
	}

	int F = in.nextInt();
	
	for(int j=0; j<F; j++){
	    int A = in.nextInt();
	    int B = in.nextInt();

	    for(int k=A; k<=B; k+=soma){
		soma = ac[B] - ac[A-1];
	    }

	    if(A == B)
		soma = energia[A];
	       
	    FastPrint.out.println(soma);
	    soma = 0;
	}

	FastPrint.out.close();
    }
}
